ncm-opennebula/src/main/perl/OpenNebula/AII.pm

```diff
@@ -0,0 +1,433 @@
+#${PMpre} NCM::Component::OpenNebula::AII${PMpost}
+
+use NCM::Component::OpenNebula::Server qw($ONEADMIN_USER);
+use Readonly;
+
+Readonly my $MINIMAL_ONE_VERSION => version->new("4.8.0");
+Readonly my $AII_OPENNEBULA_CONFIG => "/etc/aii/opennebula.conf";
+Readonly my $HOSTNAME => "/system/network/hostname";
+Readonly my $DOMAINNAME => "/system/network/domainname";
+Readonly my $MAXITER => 20;
+Readonly my $TIMEOUT => 30;
+Readonly my $ONE_DEFAULT_URL => 'http://localhost:2633/RPC2';
+Readonly my $ONE_DEFAULT_PORT => 2633;
+Readonly my $BOOT_V4 => [qw(network hd)];
+Readonly my $BOOT_V5 => [qw(nic0 disk0)];
+
+=head1 NAME
+
+C<NCM::Component::OpenNebula::AII> adds C<OpenNebula> C<VM> 
+management support.
+
+=head2 Public methods
+
+=over
+
+=item process_template_aii
+
+Detect and process C<OpenNebula> C<VM> templates.
+
+=cut
+
+sub process_template_aii
+{
+    my ($self, $config, $tt_name, $oneversion) = @_;
+
+    my $tree = $config->getElement('/')->getTree();
+    if ((defined $oneversion) and ($oneversion >= version->new("5.0.0"))) {
+        $tree->{system}->{opennebula}->{boot} = $BOOT_V5;
+        $self->verbose("BOOT section set to support OpenNebula versions >= 5.0.0");
+    } else {
+        $self->verbose("BOOT section set to support OpenNebula versions < 5.0.0");
+        $tree->{system}->{opennebula}->{boot} = $BOOT_V4;
+    };
+
+    # TBD process_template refactoring
+    my $tpl = CAF::TextRender->new(
+        $tt_name,
+        $tree,
+        relpath => 'opennebula',
+        log => $self,
+        );
+    if (!$tpl) {
+        $self->error("TT processing of $tt_name failed.", $tpl->{fail});
+        return;
+    }
+    return "$tpl";
+}
+
+=item read_one_aii_conf
+
+Reads a config file in C<.ini> style with a minimal RPC endpoint setup.
+Returns an C<OpenNebula> instance afterwards.
+
+=cut
+
+sub read_one_aii_conf
+{
+    my ($self, $data) = @_;
+
+    my $rpc = "rpc";
+
+    if (! -f $AII_OPENNEBULA_CONFIG) {
+        $self->error("No configfile $AII_OPENNEBULA_CONFIG.");
+        return;
+    }
+
+    my $config = Config::Tiny->new;
+    my $domainname = $data->getElement ($DOMAINNAME)->getValue;
+    my $hostname = $data->getElement ($HOSTNAME)->getValue;
+    my $fqdn = "${hostname}.${domainname}";
+
+    $config = Config::Tiny->read($AII_OPENNEBULA_CONFIG);
+    foreach my $section (sort keys %{$config}) {
+        $self->verbose("Found RPC section: $section");
+        my $pattern = $config->{$section}->{pattern};
+        if ($pattern and $fqdn =~ /^$pattern$/ and $rpc eq 'rpc') {
+            $rpc = $section;
+            $self->info("Match pattern in RPC section: [$rpc]");
+            last;
+        };
+    };
+    if (exists($config->{$domainname}) and $rpc eq 'rpc') {
+        $rpc = $domainname;
+        $self->info ("Detected configfile RPC section: [$rpc]");
```

Comment:
verbose is enough, mention in message that this is becuase of `domainname $domainname`